File: pdu/python/hako_srv_msgs/pdu_cdr_conv_AckEventResponse.py
from .pdu_pytype_AckEventResponse import AckEventResponse
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_AckEventResponse(writer: CdrWriter, src: AckEventResponse):
    writer.write_uint32(src.ack_code)


def cdr_body_to_py_AckEventResponse(reader: CdrReader, dst: AckEventResponse):
    dst.ack_code = reader.read_uint32()
    return dst


def py_to_cdr_AckEventResponse(src: AckEventResponse) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_AckEventResponse(writer, src)
    return writer.bytes()


def cdr_to_py_AckEventResponse(cdr_payload) -> AckEventResponse:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = AckEventResponse()
    return cdr_body_to_py_AckEventResponse(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_AckEventResponse.js
import { AckEventResponse } from './pdu_jstype_AckEventResponse.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduAckEventResponseConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { AckEventResponse } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.ack_code);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { AckEventResponse } dst
     * @returns { AckEventResponse }
     */
    static cdr_body_to_js(reader, dst) {
        dst.ack_code = reader.read_uint32();
        return dst;
    }

    /**
     * @param { AckEventResponse } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { AckEventResponse }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new AckEventResponse());
    }
}
